roles/common/tasks/fips-redhat.yml

```diff
@@ -15,3 +15,9 @@
 - name: Configure crypto policies
   shell: update-crypto-policies --set FIPS
   become: true
+  when: ansible_distribution != "Amazon"
+
+- name: Configure crypto policies
+  shell: fips-mode-setup --enable
+  become: true
+  when: ansible_distribution == "Amazon"
```

Comment:
This condition should also have ansible_major_version to identify its amazon2023 and not some older amazonlinux version